### PR TITLE
Basic template repo setup with default typescript configuration

### PR DESCRIPTION
**Pull request by lewisdaly**
Resolves: https://github.com/mojaloop/project/issues/850

This is meant to serve as a starting point for new typescript projects.

Please note: in this repo, we add `/dist` to the `.gitignore` file, so no actual compiled code is stored in the repo. Follow the conversation here: https://github.com/mojaloop/design-authority/issues/2#issuecomment-509640679

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+  command: |
+    curl -X POST \
+      $SLACK_WEBHOOK_ANNOUNCEMENT \
+      -H 'Content-type: application/json' \
+      -H 'cache-control: no-cache' \
+      -d "{
+      \"text\": \"*${CIRCLE_PROJECT_REPONAME}* - Release \`${CIRCLE_TAG}\`: https://github.com/mojaloop/${CIRCLE_PROJECT_REPONAME}/releases/tag/${CIRCLE_TAG}\"
+    }"
+
+jobs:
```
Comment (rmothilal) [inline .circleci/config.yml]: no config for license check?

File: istanbul.yml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+instrumentation:
```
Comment (rmothilal) [inline istanbul.yml]: no need for this is you have nyc

File: package.json [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+  },
+  "homepage": "https://github.com/mojaloop/<repo-name>#readme",
+  "dependencies": {},
+  "devDependencies": {
+    "@typescript-eslint/eslint-plugin": "1.11.0",
+    "@typescript-eslint/parser": "1.11.0",
+    "eslint": "6.0.1",
+    "eslint-config-prettier": "6.0.0",
+    "eslint-plugin-prettier": "3.1.0",
+    "istanbul": "1.1.0-alpha.1",
```
Comment (rmothilal) [inline package.json]: please remove if you have nyc

File: .nycrc.yaml [hunk 4]
```diff
@@ -0,0 +1,14 @@
+include:
```
Comment (rmothilal) [inline .nycrc.yaml]: please add:
temp-directory: "./.nyc_output"
check-coverage: true
per-file: true
lines: 90
statements: 90
functions: 90
branches: 90
all: true
exclude: 
  - "**/node_modules/**"

File: .nycrc.yaml [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+  - "src/**/*.ts"
+  - "test/**/*.test.ts"
```
Comment (rmothilal) [inline .nycrc.yaml]: please remove the include for test folder as we dont want out coverage including that

File: package.json [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+    "dist"
+  ],
+  "scripts": {
+    "test": "npm run test:unit | faucet",
+    "pretest": "npm run build",
+    "standard": "echo '\\033[1;33m  This project uses eslint instead of standard. Use `npm run lint` instead.'",
+    "lint": "eslint ./src/**/*.ts ./test/**/*.ts",
+    "build": "tsc",
+    "watch": "tsc -w",
+    "test:unit": "ts-node node_modules/tape/bin/tape test/unit/**/*.test.ts",
```
Comment (rmothilal) [inline package.json]: wouldnt you just use tapes here instead of ts-node node_modules/tape/bin/tape

**Comment by rmothilal** [comment]
Please include a LICENSE.md from one of the other mojaloop libraries

File: .circleci/config.yml (hunk 1 above)
Comment (lewisdaly) [reply .circleci/config.yml]: It was added later on, so no. I'll do it now.

File: .nycrc.yaml (hunk 4 above)
Comment (lewisdaly) [reply .nycrc.yaml]: I'll update this, but I've dug around a bit more into the nyc config for typescript here: https://www.npmjs.com/package/@istanbuljs/nyc-config-typescript

We can extend from the `@istanbuljs/nyc-config-typescript` package, which sets up a lot of the config for us

**Comment by lewisdaly** [comment]
@rmothilal ready to go again.

**Comment by lewisdaly** [comment]
@rmothilal I think you need to approve it..

**Comment by rmothilal** [comment]
I did

**Pull request merged**